### ISSUE TITLE
[Feature request] Return to fzf listing after playback is stopped

### ISSUE THREAD

**Issue by Kabouik** [pystardust/ytfzf#276]
Pretty much what the title says, I think it would be great if `ytfzf` showed the `fzf` listing again after the video window has been closed and the playback is stopped.

This would save users from scrapping Youtube and downloading thumbnails multiple times when they want to watch multiple videos, and also fewer keystrokes in most cases since I assume many users would either want to peek at multiple videos before choosing one, or close the terminal window after playback, in which case it wouldn't be an issue if `fzf` is still showing.

**Comment by Kabouik** [comment]
Nevermind, I just found https://github.com/pystardust/ytfzf/issues/244. Sorry for the noise. I believe it would be a nice default option though.

**Comment by Kabouik** [comment]
Woops, in fact `--search-again` is not exactly what I want. For instance if I start `ytfzf -St`, I want it to show the `-S` listing again, not prompt for a query by keyword (which would scrap Youtube and download thumbnails again anyway). Is there a way to just make the initial listing resume and show already cached results after `mpv` (or other player) is closed?

**Comment by Euro20179** [comment]
i think you might also like `-l` which after the video ends will open the fzf menu, but without making a new search query.

if you wish to have this in your config you can add `enable_loop=1`

**Comment by Kabouik** [comment]
That's perfect, thanks, and sorry for not finding about it first. Glad to see the future is already there!

It doesn't seem to go very well with `search_again=1` though. I wonder if there might be ways to make them work with each other more gracefully? Like if I start `ytfzf` with no search term (*i.e.*, if I used `-S`), then `search_again=1` could be ignored because there was no search in the first place. Or `enable_loop=1` could take priority all the time to save scrapping/download time, and the `search_again` option could be used only if `^C` or `Esc` are pressed once? I'm trying to think if there could be corner cases where one would enable both options in the configure file and not want `enable_loop` to show first, and I have not found one yet but I may be missing something.

**Comment by Euro20179** [comment]
you can add the code snippet below to your config, and it does basically what your first request asks.

```sh
on_opt_parse () {
    option="$1"
    case "$option" in
        #if the option is -S -H or -T, enable loop and disable search
        S|H|T)
            enable_loop=1
            search_again=0 ;;
    esac
}
```
I think having enable_loop supersede search_again is a case by case basis, sometimes your are making new searches constantly, like if you are listening to a bunch of different songs, but sometimes you do want loop to supersede, like if you looked up a "tips and tricks for [something]".
It would be nice to do stuff with certain keypress, but I haven't figured out how yet, pressing ctrl-c in fzf doesn't send the kill signal to ytfzf so it makes it hard to interact with.
In my config I have a somewhat complex system to make the search `!!` search the last search and ![num] search [num] search (like bash) without having to re scrape so it's slightly faster, if you want to use it you need to be on the dev branch (`ytfzf --update-unstable`).
We're thinking about adding !! in a way where all results are cached and when !! is given it shows all results from all searches in the session.

with ![num]
```sh
search_count=0

on_get_search () {
#if the search matches the ![num|!] pattern
    if printf "%s" "$1" | grep -q -Po '!(?:-?[0-9]+|!)'; then
        number="${1#!}"
        #if the 2nd part is !, set the search number to the current search count
        if [ "$number" = "!" ]; then
            number="$search_count"
        #number is going to be 1 off if negative
        elif [ "$number" -lt 0 ]; then
            number=$((search_count + number + 1))
        fi
        videos_data=$(cat /tmp/ytfzf-$number)
    fi
    search_count=$((search_count + 1))
}

on_video_data_gotten  () {
    video_data="$1"
    printf "%s" "$video_data" > /tmp/ytfzf-$search_count
}

on_exit () {
    for i in $(ls /tmp); do
        if printf "%s" "$i" | grep -qPo '^ytfzf-.*'; then
            rm /tmp/$i
        fi
    done
}
```

if you don't want ![num] you can use this snippet instead
```sh
on_get_search () {
    if [ "$1" = "!!" ]; then
        videos_data="$(cat /tmp/ytfzf-search)"
    fi
}

on_video_data_gotten  () {
    video_data="$1"
    printf "%s" "$video_data" > /tmp/ytfzf-search
}

on_exit () {
    rm /tmp/ytfzf-search
}
```

if you want it to cache everything and have !! do everything (if this doesn't work try changing `printf "%s" "$videos_data"` to `printf "%s\n" "$videos_data"`)
```sh
on_get_search () {
    if [ "$1" = "!!" ]; then
        videos_data="$(cat /tmp/ytfzf-search)"
   fi
}

on_video_data_gotten () {
    video_data="$1"
    printf "%s" "$video_data" >> /tmp/ytfzf-search
}

on_exit () {
    rm /tmp/ytfzf-search
}
```

**Comment by Kabouik** [comment]
> I think having enable_loop supersede search_again is a case by case basis, sometimes your are making new searches constantly, like if you are listening to a bunch of different songs, but sometimes you do want loop to supersede, like if you looked up a "tips and tricks for [something]".

True. I myself often listen to music by artist or album (and don't often use Youtube for that), so I never found myself in a situation where I would need multiple searches in a row more than playing multiple results of a single search in a row. So for me having `enable_loop` and `search_again` both enabled but having the former supersede the latter, and get the search prompt again only after I exit the previous results (but my understanding from your post is it is not easy) would be best, but I do understand this may depend on users.

I'll give each of your snippets a try, and will definitely move to the `development` branch, thanks a lot! I'm not sure if I should leave the issue open: there are solutions to what I was initially asking but the issue turned into how to make concurrent configuration options better work together when they should not necessarily be self exclusive.

**Comment by Euro20179** [comment]
It might be better to open a new issue,
I have an idea to make `ctrl-c` close fzf and open a search propmt, and `esc` would just close it, but we'll see, if it does work it'll probably be an option because some people (including myself) use ctrl-c to just exit fzf.

**Comment by Euro20179** [comment]
ok, I think i have come up with a good solution with #279

**Comment by pystardust** [comment]
also refer #288
